pages/api/signup.ts

```diff
@@ -2,51 +2,54 @@ import { NextApiRequest, NextApiResponse } from "next";
 
 export async function handler(req: NextApiRequest, res: NextApiResponse) {
     try {
```

Comment:
In between the "export async function handler..." and the "try {", make sure to check if the req.method == "POST"; that way it only allows post requests.